src/sentry/api/serializers/models/group.py

```diff
@@ -340,7 +340,7 @@ def serialize(self, obj, attrs, user):
 
         # If user is not logged in and member of the organization,
         # do not return the permalink which contains private information i.e. org name.
-        if user.is_authenticated() and user.get_orgs().filter(id=obj.organization.id).exists():
+        if user.is_superuser or user.is_authenticated() and user.get_orgs().filter(id=obj.organization.id).exists():
```

Comment:
So I'm not sure how you can do this easily, but you need to use `sentry.auth.superuser:is_active_superuser(request)` instead. Otherwise, this check isn't fully valid.

I'm also not great at programming, but I think you need parenthesis in here to assure the logic is correct.

Like: `if superuser or (is_authenticated and ... ):`

Reply:
hmmm maybe it makes more sense to just handle this in the frontend, and just not show the permalink at all

Reply:
That sounds bad, if we're trying to keep this more private, that suggestion means we're returning it from the API response then for everyone, and only hiding it from the UI.

An alternative is a SuperuserGroupSerializer vs GroupSerializer, which just contains this logic. Then you can choose this serializer from the API endpoint. We do stuff like this elsewhere for like, Admin*Serializer.

Reply:
no I mean if we don't have it in the response, don't try to render it in the linked issue. that would just mean superusers don't see that context, which I mean I don't know how important it is. It's just extra info in the description/summary.